README.md

```diff
@@ -10,59 +10,146 @@ A [JSON schema](api/schema.json) describing the report is also available.
 const s2 = s1.withSuite('somefile.test.js')
+s1.attach({name: 'screenshot1.png', path: './screenshot1.png'})
 s2.withTest('yay', {
   status: Status.Passed,
   duration: 123,
+  attachments: [
+    {name: 'video.mp4', path: './video.mp4'},
+    {name: 'screenshot2.png', path: './screenshot2.png'}
+  ],
 })
 s2.withTest('nay', {
   status: Status.Failed,
+  output: 'test failed',
   duration: 123,
+  attachments: [
+    {name: 'video.mp4', path: './video.mp4'},
+  ]
 })
 
 r.stringify() // returns a JSON string representing the entire test run
 // or
-r.toFile('myreport.json') // writes the JSON to a file instead
+r.toFile('my_json_report.json') // writes the JSON to a file instead
+// or
+r.toJUnitFile('my_junit_report.xml') // converts and writes the result to a JUnit file
 ```
 
 The resulting JSON of the above example is:
 
 ```json
 {
   "status": "failed",
-  "attachments": [],
+  "attachments": [
+    {
+      "name": "screenshot.png",
+      "path": "./screenshot.png"
+    }
+  ],
   "suites": [
     {
       "name": "somegroup",
       "status": "failed",
+      "metadata": {},
       "suites": [
         {
           "name": "somefile.test.js",
           "status": "failed",
+          "metadata": {},
           "suites": [],
           "attachments": [],
           "tests": [
             {
               "name": "yay",
               "status": "passed",
-              "startTime": "2021-11-09T21:10:59.550Z",
               "duration": 123,
-              "attachments": []
+              "startTime": "2023-06-16T22:16:10.595Z",
+              "attachments": [
+                {
+                  "name": "video.mp4",
+                  "path": "./video.mp4"
+                },
+                {
+                  "name": "screenshot2.png",
+                  "path": "./screenshot2.png"
+                }
+              ],
+              "metadata": {}
             },
             {
               "name": "nay",
               "status": "failed",
-              "startTime": "2021-11-09T21:10:59.550Z",
               "duration": 123,
-              "attachments": []
+              "output": "test failed",
+              "startTime": "2023-06-16T22:16:10.595Z",
+              "attachments": [
+                {
+                  "name": "video.mp4",
+                  "path": "./video.mp4"
+                }
+              ],
+              "metadata": {}
             }
           ]
         }
       ],
-      "attachments": [],
+      "attachments": [
+        {
+          "name": "screenshot1.png",
+          "path": "./screenshot1.png"
+        }
+      ],
       "tests": []
     }
-  ]
+  ],
+  "metadata": {}
 }
 ```
+
+The resulting JUnit file of the above example is:
+```
+<testsuites status="failed">
+  <testsuite name="somegroup" status="failed">
+    <properties>
+      <property name="attachment" value="screenshot1.png">./screenshot1.png</property>
+      <property name="metadata">
+        <metadata/>
```

Comment:
Pondering out loud: Isn't `property` effectively metadata?
If metadata is an arbitrary map of key-value pairs, then it can be directly broken out into many `property` entries:
```xml
<property name="key" value="value" />
<property name="key1" value="value2" />
<property name="key2" value="value3" />
```

Reply:
I'm fine with breaking them using suggested format as I checked almost all of metadata are just key-value pairs.
```
<property name="key" value="value" />
```
While I'm not sure about `metadata` field here as it is not a predictable data structure. I tried
```
<property name="metadata"> JSON.stringify(metadata) </property>
```
But it looks odd.